mParticle-Apptimize/MPKitApptimize.m

```diff
@@ -51,6 +53,22 @@ - (MPKitExecStatus*) makeStatus:(MPKitReturnCode)code {
             returnCode:code];
 }
 
+#pragma mark - MPKitApptimize methods
+
++ (void)setPilotTargetingID:(NSString*) pilotTargetingID {
+    void(^block)(void) = ^{
+        _pilotTargetingID = pilotTargetingID;
+        if ([[MParticle sharedInstance] isKitActive:[[self class] kitCode]]) {
+            [Apptimize setPilotTargetingID: pilotTargetingID];
```

Comment:
I don't think this block needs to be called on the main thread. The documentation doesn't specify that it needs to be.

The call to `startApptimizeWithApplicationKey` does specify it needs to be called on the main thread.

Reply:
I have surrounded read/write access to _pilotTargetingID on the main thread.

MParticle isKitActive does not necessarily specify to be called on the main thread but doing so is fine.

For `setUserIdentity:` below I believe it is called on the main thread.

Reply:
not 100% sure what you meant, but i think we should probably remove the thread switching for block execution if it's not required — otherwise people might assume main thread execution is an undocumented requirement